### PR TITLE
Let me prevent permissions changes

### PR DESCRIPTION
**Pull request by jonkeane**
This will prevent various CI errors I'm seeing with arrowbench, since we create a cache in a temp dir, but that temp dir is not able to be cleaned up. https://github.com/voltrondata-labs/arrowbench/actions/runs/3421285685/jobs/5697219852#step:11:74

### CODE REVIEW

File: datalogistik/util.py [hunk 1]
```diff
@@ -18,32 +18,39 @@
...
-from .tpc_builders import DBGen, DSDGen
+
+
+def getenv_bool(name: str, default: bool = False) -> bool:
+    return os.getenv(name, str(default)).lower() in ("true", "t", "yes", "y", "1")
 
 
 # Set file permissions of given path to readonly
 def set_readonly(path):
+    if getenv_bool("DATALOGISTIK_NO_PERMISSIONS_CHANGE"):
```
Comment (austin3dickey) [inline datalogistik/util.py:38]: Could you add this to README?

**Review by austin3dickey — APPROVED** [review]
Nice

File: datalogistik/generate_dataset.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+from .log import log
+from .table import Table
+from .tpc_builders import DBGen, DSDGen
+from .util import clean_cache_dir
+
+
+# Generate a dataset by calling one of the supported external generators
+# TODO: Generator output cannot be used as dataset output directly, because of the
+# trailing columns.
+def generate_dataset(dataset):
```
Comment (joosthooz) [inline datalogistik/generate_dataset.py:27]: Nice, now it should be possible to add type hints here without creating a circular dependency!
(or we could do that in #118 after merging this)

File: datalogistik/tpc_builders.py [hunk 3]
```diff
@@ -173,10 +173,9 @@ def create_dataset(
                 # Not all tables will have enough rows for the full nr of partitions
                 if old_file.exists():
                     shutil.move(old_file, new_file)
-                    # reset permissions to read-only
-                    os.chmod(new_file, 0o444)
                     log.debug(f"Created {new_file}")
-            os.chmod(table_output_dir, 0o555)
+            # reset permissions to read-only
+            set_readonly_recurse(table_output_dir)
```
Comment (joosthooz) [inline datalogistik/tpc_builders.py:178]: Nice, another circular dependency removed :+1:

**Review by joosthooz — APPROVED** [review]

File: datalogistik/generate_dataset.py (hunk 2 above)
Comment (jonkeane) [reply datalogistik/generate_dataset.py:27]: Let's do it in #118, if you don't mind

File: datalogistik/util.py (hunk 1 above)
Comment (jonkeane) [reply datalogistik/util.py:38]: Done, there wasn't any section there about permissions (and this is really a pretty hidden feature regardless), but it is good to document things like this

**Pull request merged**